datadog_checks_base/datadog_checks/base/checks/windows/perf_counters/counter.py

```diff
@@ -0,0 +1,516 @@
+# (C) Datadog, Inc. 2021-present
+# All rights reserved
+# Licensed under a 3-clause BSD style license (see LICENSE)
+import re
+from collections import defaultdict
+
+import pywintypes
+import win32pdh
+
+from ....errors import ConfigTypeError, ConfigValueError
+from .transform import NATIVE_TRANSFORMERS, TRANSFORMERS
+from .utils import construct_counter_path, get_counter_value
+
+
+class PerfObject:
+    def __init__(self, check, connection, name, config, use_localized_counters, tags):
+        self.check = check
+        self.logger = check.log
+        self.connection = connection
+        self.use_localized_counters = config.get('use_localized_counters', use_localized_counters)
+        self.tags = tags
+
+        # The name of the performance object e.g. System
+        self.name = name
+
+        # The prefix of all metrics
+        self.metric_prefix = config.get('name', '')
+        if not isinstance(self.metric_prefix, str):
+            raise ConfigTypeError(f'Option `name` for performance object `{self.name}` must be a string')
+        elif not self.metric_prefix:
+            raise ConfigValueError(f'Option `name` for performance object `{self.name}` is required')
+
+        # The tag name used for any instances
+        self.tag_name = config.get('tag_name', '')
+        if not isinstance(self.tag_name, str):
+            raise ConfigTypeError(f'Option `tag_name` for performance object `{self.name}` must be a string')
+
+        include_patterns = config.get('include', [])
+        if not isinstance(include_patterns, list):
+            raise ConfigTypeError(f'Option `include` for performance object `{self.name}` must be an array')
+        elif not include_patterns:
+            self.include_pattern = None
+        else:
+            for i, pattern in enumerate(include_patterns, 1):
+                if not isinstance(pattern, str):
+                    raise ConfigTypeError(
+                        f'Pattern #{i} of option `include` for performance object `{self.name}` must be a string'
+                    )
+
+            self.include_pattern = re.compile('|'.join(include_patterns))
+
+        exclude_patterns = config.get('exclude', [])
```

Comment:
should we log any duplicate values that are in `include` and `exclude` lists?

Reply:
I think not worth it